### PR TITLE
Use mixed Horner scheme for better parallel instruction decoding

### PR DESCRIPTION
**Pull request by MaxGraey**
Bench comparison:
https://webassembly.studio/?f=qnm9vtskydl
On Firefox this improvements speed-up `exp` and `expm1` on 15%-20% on Firefox.

**TODO**
- [x] add more tests for making sure we don't loose precision

### CODE REVIEW

File: std/assembly/math.ts [hunk 1]
```diff
@@ -423,7 +423,9 @@ export namespace NativeMath {
       hi = x;
     } else return 1.0 + x;
     var xx = x * x;
-    var c = x - xx * (P1 + xx * (P2 + xx * (P3 + xx * (P4 + xx * P5))));
+    // var c = x - xx * (P1 + xx * (P2 + xx * (P3 + xx * (P4 + xx * P5))));
+    var xx2 = xx * xx;
```
Comment (dcodeIO) [inline std/assembly/math.ts]: Should either make it `xx` and `xxxx` or `x_2` and `x_4` to avoid confusion.

Comment (MaxGraey) [reply std/assembly/math.ts]: What about `xp2` and `xp4`? I want keep names shortest as much as possible

Comment (dcodeIO) [reply std/assembly/math.ts]: Sure, that's fine as well

Comment (dcodeIO) [reply std/assembly/math.ts]: Though, the other two resemble a mathematical formula a bit more hmm ^^

Comment (MaxGraey) [reply std/assembly/math.ts]: I see better way. Using `s` (square) and `q` (quadruple)

Comment (MaxGraey) [reply std/assembly/math.ts]: And this notation already used in this [line](https://github.com/AssemblyScript/assemblyscript/pull/311/files#diff-4f1e427ed70aece4d8c8162e9b85ad54L466)

**Comment by dcodeIO** [comment]
Thanks :)

**Pull request merged**